docs/examples/Naive_Classification.py

```diff
@@ -0,0 +1,188 @@
+np.random.seed(1991)
+
+gt = []
+nsteps=100
+
+for step in range(nsteps):
+    gt.append(State(StateVector([0, 0, 1])))  # bicycle, car, bus
+
+
+# %%
+# Modelling state transition
+# --------------------------
+# We require a model of the state transition in order to track the target's classification.
+# This will be a simple linear map, represented by a :math:`3\times 3` matrix with no noise added.
+# We might not know that the target does not switch classification, so the matrix should not simply
+# be the identity matrix.
+#
+# .. math::
+#       T = \begin{bmatrix}
+#  		P(Bicycle_{t}|Bicycle_{t-1}) & P(Bicycle_{t}|Car_{t-1}) & P(Bicycle_{t}|Bus_{t-1}) \\
+#  		P(Car_{t}|Bicycle_{t-1}) & P(Car_{t}|Car_{t-1}) & P(Car_{t}|Bus_{t-1}) \\
+#  		P(Bus_{t}|Bicycle_{t-1}) & P(Bus_{t}|Car_{t-1}) & P(Bus_{t}|Bus_{t-1})
+#  	    \end{bmatrix}
+#
+# The transitioned state is given by
+# :math:`f(X_{t-1})_{j} = T_{ij}{X_{t-1}}_{i} = T^T_{ji}{X_{t-1}}_{i}`.
+# Therefore, we take the transpose :math:`f(X_{t-1}) = T^{T}X_{t-1}` when transitioning the state.
+
+F = np.array([[0.8, 0.1, 0.1],
+              [0.1, 0.8, 0.1],
+              [0, 0.1, 0.9]])
+
+
+def transit(state):
+    return F.T @ state.state_vector
+
+
+# %%
+# Modelling observations
+# ----------------------
+# Observations of the target may be incorrect, and affected by various input from a sensor.
+# We simply model an observer of the target's size, with an understanding of the underlying
+# distribution of bicycle, car and bus targets that are/could be small, medium or large.
+# This is defined in the emission matrix, utilised by the observer model to determine a measurement
+# which just says whether the target is small, medium or large. Similar to the state space
+# categories, this is represented by a detection state vector with 3 components, either 0 or 1.
+# To determine which classification is observed, we randomly sample from the multinomial
+# distribution defined by the row of the emission matrix corresponding to the most likely state of
+# the target.
+#
+# .. math::
+#       E = \begin{bmatrix}
+# 		P(Small | Bicycle) & P(Medium | Bicycle) & P(Large | Bicycle)\\
+# 		P(Small | Car) & P(Medium | Car) & P(Large | Car)\\
+# 		P(Small | Bus) & P(Medium | Bus) & P(Large | Bus)
+# 	    \end{bmatrix}
+#
+
+import scipy
+
+E = np.array([[0.89, 0.1, 0.01],
+              [0.3, 0.3, 0.4],
+              [0.1, 0.3, 0.6]])
+
+# ie. 89% of bicycles are small, 10% are medium, 1% are large.
+
+
+def observe(state):
+    x = state.state_vector
+
+    row_num = np.argmax(x)
+
+    row = E[row_num]
+
+    sample = _sample(row)
+
+    return StateVector(sample)
+
+
+def _sample(row):
+    rv = scipy.stats.multinomial(n=1, p=row)
+    return rv.rvs(size=1, random_state=None)
+
+
+# %%
+
+from stonesoup.types.detection import Detection
+
+observations = []
+for i in range(0, nsteps):
+    observations.append(Detection(observe(gt[i])))
+
+
+# %%
+# Tracking classification
+# -----------------------
```

Comment:
This merits a bit more explanation. (See https://en.wikipedia.org/wiki/Forward_algorithm for a start.) In particular it would be good to make connection between the predict-update recursion in other Stone Soup filters and the transition-emission probabilities here.